### PR TITLE
util-linux: add linux-only dependency, make keg-only only on mac

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Review by SMillerDev — DISMISSED** [review]

**Comment by iMichka** [comment]
I need to fix the audit first

File: Formula/util-linux.rb [hunk 1]
```diff
@@ -20,14 +20,17 @@ class UtilLinux < Formula
...
+    keg_only "macOS provides the uuid.h header"
+  end
 
   depends_on "autoconf" => :build
   depends_on "automake" => :build
   depends_on "libtool" => :build
   depends_on "pkg-config" => :build
   depends_on "gettext"
 
+  uses_from_macos "bison" => :build
```
Comment (bayandin) [inline Formula/util-linux.rb:33]: Do we need `bison` to call `./autogen.sh` only? If so, could you please add it to a patch comment below (to remove this dependency in the next release)

File: Formula/util-linux.rb [hunk 2]
```diff
@@ -20,14 +20,17 @@ class UtilLinux < Formula
     sha256 mojave:        "de7bfed47b70d497e2406b7813b966aad7a0436e6fd129d4e12f5df5757e3ef9"
   end
 
-  keg_only "macOS provides the uuid.h header"
+  on_macos do
+    keg_only "macOS provides the uuid.h header"
+  end
```
Comment (MikeMcQuaid) [inline Formula/util-linux.rb:25]:
```suggestion
  keg_only :shadowed_by_macos, "macOS provides the uuid.h header"
```

File: Formula/util-linux.rb (hunk 1 above)
Comment (iMichka) [reply Formula/util-linux.rb:31]: I can't move it because our style audit. I added it because the build complained that it needed bison.

**Comment by alebcay** [comment]
While we're here, would it be worth considering wrapping the patch in an `on_macos` block and re-visiting the `configure` arguments (e.g. conditionally disable components on macOS if they don't build, enable the ones that shadow macOS since this is becoming `keg_only`)?

**Review by dawidd6 — DISMISSED** [review]

**Comment by iMichka** [comment]
> wrapping the patch in an on_macos block

Done

> re-visiting the configure arguments (e.g. conditionally disable components on macOS if they don't build, enable the ones that shadow macOS since this is becoming keg_only)?

The formula was already keg_only, I am just changing it from a plain string to shadowed_by_macos. Else I need to put the `keg_only` in an `on_linux` block but this is not accepted in Homebrew/brew.

It's a tricky situation I guess because some utils are shadowing macOS ones, but not all.

File: Formula/util-linux.rb [hunk 3]
```diff
@@ -20,62 +20,64 @@ class UtilLinux < Formula
...
+    depends_on "pkg-config" => :build
 
-  # Fix build for MacOS
-  # Remove in the next release
-  # Also remove autoconf/automake/libtool/pkg-config dependencies and autogen.sh call
-  patch do
-    url "https://github.com/karelzak/util-linux/commit/71ba2792ab3f96b5f5d5d3b0a68d35ecfd0f93a2.patch?full_index=1"
-    sha256 "bc5188d3f41a7f248ba622f51c8ab8fed0e05355cbe20a5d3b02bbc274e2c7b4"
+    # Fix build for MacOS
+    # Remove in the next release
```
Comment (MikeMcQuaid) [inline Formula/util-linux.rb:37]:
```suggestion
    # Remove in >2.36.2
```

**Review by MikeMcQuaid — DISMISSED** [review]

**Review by dawidd6 — APPROVED** [review]

**Pull request merged**